apps/E2E/src/Checkbox/specs/Checkbox.spec.win.ts

```diff
@@ -47,36 +63,57 @@ describe('Checkbox Functional Testing', () => {
   beforeEach(async () => {
     await CheckboxPageObject.scrollToTestElement();
 
-    await CheckboxPageObject.toggleCheckboxToUnchecked();
+    await CheckboxPageObject.toggleCheckbox(false);
   });
 
-  it("Click on a Checkbox -> Validate it toggles correctly AND calls the user's onChange", async () => {
+  it("Click the primary checkbox. Validate that the Checkbox toggles correctly AND calls the user's onChange() callback.", async () => {
     /* Validate the Checkbox is initially toggled OFF */
-    await expect(await CheckboxPageObject.isCheckboxChecked()).toBeFalsy();
+    await expect(await CheckboxPageObject.isCheckboxChecked()).toBeFalsy('The primary checkbox should initially be toggled off.');
 
     /* Click on the Checkbox to toggle on */
-    await CheckboxPageObject.clickComponent();
-    await CheckboxPageObject.waitForCheckboxChecked(PAGE_TIMEOUT);
-
-    await expect(await CheckboxPageObject.didOnChangeCallbackFire()).toBeTruthy();
+    await CheckboxPageObject.click(CheckboxPageObject._primaryComponent);
 
     /* Validate the Checkbox is toggled ON */
-    await expect(await CheckboxPageObject.isCheckboxChecked()).toBeTruthy();
+    await expect(
+      await CheckboxPageObject.waitForCheckboxToggle(true, 'The primary checkbox should have been toggled on via click.'),
+    ).toBeTruthy();
 
-    await CheckboxPageObject.clickComponent();
+    await expect(
+      await CheckboxPageObject.didOnChangeCallbackFire('Primary checkbox failed to fire onChange() callback via click.'),
+    ).toBeTruthy();
+
+    await CheckboxPageObject.click(CheckboxPageObject._primaryComponent);
 
     /* Validate the Checkbox is toggled OFF */
-    await expect(await CheckboxPageObject.isCheckboxChecked()).toBeFalsy();
+    await expect(
+      await CheckboxPageObject.waitForCheckboxToggle(false, 'The primary checkbox should have been toggled off via click.'),
```

Comment:
@lawrencewin Can you confirm - If this fails, does console produce the error message from `waitForCheckboxToggle()` or the error message from this `expect()`?

Reply:
With waiters, the timeout message passed into `browser.waitFor()` always goes first.